src/track/serato/markers.cpp

```diff
@@ -0,0 +1,229 @@
+    }
+
+    // Parse End Position
+    int endPosition = -1;
+    if (endPositionSet == 0x7F) {
+        // End position not set
+        if (endPositionRaw != 0x7F7F7F7F) {
+            qWarning() << "Parsing SeratoMarkersEntry failed:"
+                       << "endPosition != 0x7F7F7F7F";
+
+            return nullptr;
+        }
+    } else {
+        endPosition = static_cast<int>(endPositionRaw);
+    }
+
+    // Make sure that the unknown (and probably unused) bytes have the expected value
+    if (strncmp(buffer, "\x00\x7F\x7F\x7F\x7F\x7F", sizeof(buffer)) != 0) {
+        qWarning() << "Parsing SeratoMarkersEntry failed:"
+                   << "Unexpected value at offset 10";
+        return nullptr;
+    }
+
+    if (stream.status() != QDataStream::Status::Ok) {
+        qWarning() << "Parsing SeratoMarkersEntry failed:"
+                   << "Stream read failed with status" << stream.status();
+        return nullptr;
+    }
+
+    SeratoMarkersEntryPointer pEntry = SeratoMarkersEntryPointer(new SeratoMarkersEntry(
+            startPosition,
+            endPosition,
+            color,
+            type,
+            isLocked));
+    qDebug() << "SeratoMarkersEntry" << *pEntry;
+    return pEntry;
+}
+
+bool SeratoMarkers::parse(SeratoMarkers* seratoMarkers, const QByteArray& data) {
+    QDataStream stream(data);
+    stream.setVersion(QDataStream::Qt_5_0);
+    stream.setByteOrder(QDataStream::BigEndian);
+
+    quint16 version;
+    stream >> version;
+    if (version != kVersion) {
+        qWarning() << "Parsing SeratoMarkers_ failed:"
+                   << "Unknown Serato Markers_ tag version";
+        return false;
+    }
+
+    quint32 numEntries;
+    stream >> numEntries;
+
+    char buffer[kEntrySize];
+    QList<SeratoMarkersEntryPointer> entries;
+    for (quint32 i = 0; i < numEntries; i++) {
+        if (stream.readRawData(buffer, sizeof(buffer)) != sizeof(buffer)) {
+            qWarning() << "Parsing SeratoMarkersEntry failed:"
+                       << "unable to read entry data";
+            return false;
+        }
+
+        QByteArray entryData = QByteArray(buffer, kEntrySize);
+        SeratoMarkersEntryPointer pEntry = SeratoMarkersEntryPointer(
+                SeratoMarkersEntry::parse(entryData));
+        if (!pEntry) {
+            qWarning() << "Parsing SeratoMarkers_ failed:"
+                       << "Unable to parse entry!";
+            return false;
+        }
+        entries.append(pEntry);
+    }
+
+    quint32 trackColorRaw;
+    stream >> trackColorRaw;
+    QRgb trackColor = colorToRgb(trackColorRaw);
+
+    if (stream.status() != QDataStream::Status::Ok) {
+        qWarning() << "Parsing SeratoMarkers_ failed:"
+                   << "Stream read failed with status" << stream.status();
+        return false;
+    }
+
+    seratoMarkers->setEntries(entries);
+    seratoMarkers->setTrackColor(trackColor);
+
+    return true;
+}
+
+QByteArray SeratoMarkers::data() const {
```

Comment:
data() gives be the impression that it is a small function returning a pointer to the internal data. 
Can we rename it to compose() or something?

Reply:
How about `dump()`?